### PR TITLE
List files can be null

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Only one and first issue via google dev console:
If it is not a folder or I/O Exception, listFiles() returns null.

Signed-off-by: tobiaskaminsky <tobias@kaminsky.me>

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
👍

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky should this be backported?

**Comment by tobiasKaminsky** [comment]
This should not happen at all, and occurs <5x, so not worth a backport.

**Comment by AndyScherzinger** [comment]
cc @mario / @przybylski for review :)

File: src/main/java/com/owncloud/android/utils/FileStorageUtils.java [hunk 1]
```diff
@@ -263,7 +263,7 @@ public int compare(OCFile o1, OCFile o2) {
      * @return Size in bytes
      */
     public static long getFolderSize(File dir) {
-        if (dir.exists()) {
+        if (dir.exists() && dir.isDirectory() && dir.listFiles() != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/utils/FileStorageUtils.java]: Can we please store dir.listFiles() in a variable and use it two lines down? Thanks!

**Comment by AndyScherzinger** [comment]
:+1: fine by me

@mario CRs have been implemented

**Comment by mario** [comment]
Merge

**Pull request merged**